pkg/ui/workspaces/cluster-ui/src/api/nodesApi.ts

```diff
@@ -9,12 +9,57 @@
 // licenses/APL.txt.
 
 import { cockroach } from "@cockroachlabs/crdb-protobuf-client";
+import { useMemo } from "react";
+import useSWR from "swr";
 
 import { fetchData } from "src/api";
+import { getRegionFromLocality } from "src/store/nodes";
 
-const NODES_PATH = "_status/nodes";
+import { NodeID, StoreID } from "../types/clusterTypes";
+
+const NODES_PATH = "_status/nodes_ui";
 
 export const getNodes =
   (): Promise<cockroach.server.serverpb.NodesResponse> => {
     return fetchData(cockroach.server.serverpb.NodesResponse, NODES_PATH);
   };
+
+export const useNodeStatuses = () => {
+  const { data, isLoading, error } = useSWR(NODES_PATH, getNodes, {
+    revalidateOnFocus: false,
+  });
+
+  const nodeIDToRegion = useMemo(() => {
+    const result: Record<NodeID, string> = {};
+    if (!data) {
+      return result;
+    }
+    data.nodes.forEach(ns => {
+      result[ns.desc.node_id as NodeID] = getRegionFromLocality(
+        ns.desc.locality,
+      );
+    });
+    return result;
+  }, [data]);
+
+  const storeIDToNodeID = useMemo(() => {
```

Comment:
Why do we need two separate useMemos here? Can the generation of both nodeIdToRegion and storeIdToNodeId be done in a single iteration of data.nodes?

Reply:
We could do that - I find it's sometimes a little cleaner to separate but we can consolidate here. Done.
Also, I think you meant to leave this comment and the one above on https://github.com/cockroachdb/cockroach/pull/130958. This PR should only focus on the latest commit.